src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,132 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>>& generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(generators)
+    , mIndex(0)
+  {
+    for (std::size_t i = 0; i < mGenerators.size(); ++i)
+      assert(mGenerators[i]->getStateSpace() == mStateSpace);
```

Comment:
Please note that `assert` is ignored in release mode. This code will not be built in release mode since the for-statement is empty.

We could either:
```cpp
#ifndef NDEBUG // debug mode
  for (const auto& generator : mGenerators)
    assert(generator->getStateSpace() == mStateSpace);
#endif
```
or
```cpp
  for (const auto& generator : mGenerators)
  {
    assert(generator->getStateSpace() == mStateSpace);
  }
```

I slightly prefer former version. :smile: